packages/data/addon/services/cardstack-data.js

```diff
@@ -122,13 +122,14 @@ export default Service.extend({
   },
 
   getCardMeta(card, attribute) {
+    let hasId = !!card.id;
     if (attribute === 'human-id') {
       let humanType = getType(card)
         .split('-')
-        .map((s) => capitalize(s))
+        .map((s) => hasId ? capitalize(s) : s)
         .join(' ');
-      return [`${humanType} `, card.id]
-        .join('#');
+
+      return card.id ? [`${humanType} `, card.id].join('#') : `New ${humanType}`;
```

Comment:
`#` should only be used if you know for sure the thing you are displaying is an ordinal number--otherwise it's just confusing.

Reply:
That's a good point but then how should we distinguish between two resources that have the same type? You'd see multiple "Comment: Body" labels and furthermore they might not even be rendered on screen to use the overlay for the visual telling apart.

One idea I had is to pass in an index to this method (well, to the entry point to this method as I recall this is not called directly) but that would increase the surface area for this method (and of all the methods on the call chain) and would make the caller too smart.